Library/Homebrew/cmd/info.rb

```diff
@@ -401,6 +419,69 @@ def info_cask(cask)
 
         Cask::Info.info(cask, args:)
       end
+
+      sig { params(formulae: T::Array[Formula], casks: T::Array[Cask::Cask]).void }
+      def print_sizes(formulae: [], casks: [])
+        formulae_to_process = if formulae.present?
+          formulae
+        elsif args.formulae? || (!args.casks? && args.no_named?)
+          Formula.installed
+        else
+          []
+        end
```

Comment:
We can probably simplify this to
```suggestion
        formulae_to_process = if args.formulae? || (!args.casks? && args.no_named?)
          Formula.installed
        else
          formulae
        end
```
Though this doesn't work correctly for
```
brew info --sizes --formulae hello awk
```
but neither did the original code.

Reply:
This also does not work in the case of when specific formulae are passed e.g. `brew info --sizes --formulae hello awk` as the condition will evaluate to false and the formulae array will become overwritten to `nil` instead of maintaining the original array of passed in formulae. The original code provided will handle this case however, if you'd like I could change it to a simple one-liner to capture the full logic 
`formulae_to_process = formulae.presence || (args.formulae? || (!args.casks? && args.no_named?) ? Formula.installed : [])`